controllers/upgrade_handlers.go

```diff
@@ -19,38 +19,134 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// BackupPhase defines the phase of backup
+type BackupPhase string
+
+// Constants for backup phase
+const (
+	BackupPending          BackupPhase = "BackupPending"
+	BackupFailedValidation BackupPhase = "BackupFailedValidation"
+	BackupFailed           BackupPhase = "BackupFailed"
+	BackupCompleted        BackupPhase = "BackupCompleted"
+	BackupInProgress       BackupPhase = "BackupInProgress"
+	BackupSkipped          BackupPhase = "BackupSkipped"
+	BackupRequested        BackupPhase = "BackupRequested"
+)
+
+// handleUpgrade orchestrate main upgrade steps
+// nolint:unparam
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	// todo: as a first pass we are catching and error and simply not requeue-ing. Check pod logs for assistance.
+	// todo: when hardening the system consider leveraging .status.condition to let users know if things failed/skipped
+
+	/*
+			e.g "substatus"
+			newCondtion := metav1.Condition{
+				Type:   fmt.Sprintf("%s%s", utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade), backuprestore.BackupRequested),
+				Status: metav1.ConditionTrue,
+				Reason: string(backuprestore.BackupRequested),
+			}
+			meta.SetStatusCondition(&ibu.Status.Conditions, newCondtion)
+
+		    output may look like:
+		  - lastTransitionTime: "2023-11-16T16:08:55Z"
+			message: ""
+			reason: BackupRequested
+			status: "True"
+			type: UpgradeInProgressBackupRequested
+	*/
+
+	// log and update status to in-progress
+	r.Log.Info("Starting handleUpgrade")
+
+	// backup with OADP
+	r.Log.Info("Starting backup")
+	reqBackup, backupCRs, err := checkIfBackupRequested(r.BackupRestore, ibu.Spec.OADPContent, ctx)
+	if err != nil {
+		r.Log.Error(err, "")
+		return doNotRequeue(), nil
+	}
+	if reqBackup {
+		// validate if backup CR is not malformed
+		err := backuprestore.ValidateBackupCRs(ctx, r.BackupRestore, backupCRs)
+		if err != nil {
+			r.Log.Error(err, "")
+			return doNotRequeue(), nil
```

Comment:
Are these errors that can be corrected by user? If they are, we would want to requeue with an interval.

Reply:
Yeah! But yesterday during we return regardless if can be fixed or not?

Reply:
See Angie's comment too. We either
- keep this upgrade alive by requeuing and maybe updating in progress condition message/reason only, or
- give it up by doNotRequeue and updating the conditions to false.

The only exception is an error that can only be corrected by updating IBU CR spec. In that case, we can return doNotRequeue while keeping the upgrade still in progress.

Reply:
Updated as discussed. Following the more traditional path